src/basic/path-util.c

```diff
@@ -671,6 +672,11 @@ int find_executable_full(const char *name, bool use_path_envvar, char **ret_file
         if (!p)
                 p = DEFAULT_PATH;
 
+        if (exec_search_path) {
+                exec_search_path_prefix = strjoina(exec_search_path, ":");
+                p = strjoina(exec_search_path_prefix, p);
+        }
```

Comment:
we probably should not use stack allocation for this string, given that part of the string is controllobale via env var, i.e. externally. just as a safety precaution